jetbrains-core/tst/software/aws/toolkits/jetbrains/services/sqs/toolwindow/PollMessagePaneTest.kt

```diff
@@ -45,10 +41,26 @@ class PollMessagePaneTest : BaseCoroutineTest() {
             tableModel.waitForModelToBeAtLeast(1)
         }
 
-        assertThat(tableModel.items.first().messageId()).isEqualTo("XYZ")
-        assertThat(tableModel.items.first().body()).isEqualTo("ABC")
-        assertThat(tableModel.items.first().attributes().getValue(MessageSystemAttributeName.SENDER_ID)).isEqualTo("1234567890:test1")
-        assertThat(tableModel.items.first().attributes().getValue(MessageSystemAttributeName.SENT_TIMESTAMP)).isEqualTo("111111111")
+        assertThat(tableModel.items.first()).isEqualTo(message)
+    }
+
+    @Test
+    fun `Messages are de-duplicated by id`() {
```

Comment:
This uses custom IDs in each message, what is causing it to test the de-dupe logic?

Reply:
not sure what you mean `aMessage()` creates a new message with a randomly generated id by calling `aString()`

Reply:
Yes, but this is called are de-duplicated by ID,

If the two messages have different IDs, this test isnt actually testing the .distincyBy